xscen/aggregate.py

```diff
@@ -334,36 +335,72 @@ def spatial_mean(
     xarray.Dataset.mean, xarray.Dataset.interp, xesmf.SpatialAverager
     """
     kwargs = kwargs or {}
+    if method == "mean":
+        warnings.warn(
+            "xs.spatial_mean with method=='mean' is deprecated and will be abandoned in a future release. Use method=='cos-lat' instead for a more robust but similar method.",
+            category=DeprecationWarning,
+            stacklevel=2,
+        )
+    elif method == "interp_coord":
+        warnings.warn(
+            "xs.spatial_mean with method=='interp_coord' is deprecated. Use method=='interp_centroid' instead.",
+            category=DeprecationWarning,
+            stacklevel=2,
```

Comment:
```suggestion
            category=FutureWarning,
        )
    elif method == "interp_coord":
        warnings.warn(
            "xs.spatial_mean with method=='interp_coord' is deprecated. Use method=='interp_centroid' instead.",
            category=FutureWarning,
```
From python's doc:

**DeprecationWarning** : Base category for warnings about deprecated features when those warnings are intended for other Python developers (ignored by default, unless triggered by code in __main__).
**FutureWarning** : Base category for warnings about deprecated features when those warnings are intended for end users of applications that are written in Python.

Also, `stacklevel=2` lands in the `parse_config` wrapper. Without it (=1), the warning is shown was stemming from inside the function. And with 3, it stems from call of `spatial_mean` (like the workflow or the notebook cell itself).